src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/CLinker.java

```diff
@@ -54,11 +109,298 @@ private CSupport() {}
      * @throws IllegalAccessError if the runtime property {@code foreign.restricted} is not set to either
      * {@code permit}, {@code warn} or {@code debug} (the default value is set to {@code deny}).
      */
-    public static ForeignLinker getSystemLinker() {
-        Utils.checkRestrictedAccess("CSupport.getSystemLinker");
+    static CLinker getInstance() {
+        Utils.checkRestrictedAccess("CLinker.getInstance");
         return SharedUtils.getSystemLinker();
     }
 
+    /**
+     * Obtain a method handle which can be used to call a given native function.
```

Comment:
This feels a bit thin. I would replace with:
```
Obtain a foreign method handle, with given type, which can be used to call a target foreign function at a given address and featuring a given function descriptor.
```

Let's also consider renaming `downcallHandle` to `foreignHandle`